elyra/util/git.py

```diff
@@ -0,0 +1,84 @@
+# limitations under the License.
+#
+
+from github import Github, GithubException
+from traitlets.config import LoggingConfigurable
+from typing import Optional
+
+
+class GithubClient(LoggingConfigurable):
+
+    def __init__(self,
+                 token: str,
+                 repo: str,
+                 branch: Optional[str] = None,
+                 server_url: Optional[str] = None):
+        """
+        Creates a Github Client for Elyra
+        :param token: Personal Access Token for use with Github
+                      See https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+        :param repo: Github Repository to use. Use Form : [Github Username/Org]/[Repository Name]
+                     e.g. elyra/examples
+        :param branch: Branch in the 'repo' to use. If not provided, this will use the default branch configured in the
+                       target repository
+        :param server_url:  Github API endpoint to use for the client. This is can be an Enterprise
+                          Github instance. By default the client will attempt to connect to the main Github API at
+                          https://api.github.com'
+        """
+
+        super().__init__()
+
+        if server_url:
+            self.client = Github(login_or_token=token, base_url=server_url)
+        else:
+            self.client = Github(login_or_token=token)
+
+        try:
+            self.repo = self.client.get_repo(repo)
+        except GithubException as e:
+            self.log.error(f'Error accessing repository {repo}: ' + str(e))
+            raise RuntimeError(f'Error accessing repository {repo}: ', e)
+
+        self.branch = branch
+
+    def upload_dag(self,
+                   pipeline_filepath: str,
+                   pipeline_name: str) -> None:
+        """
+        Push a DAG to a remote Github Repository
+        :param pipeline_filepath: filepath to the location of the DAG in the local filesystem
+        :param pipeline_name: the name of the file to be created in the remote Github Repository
+        :return:
+        """
+
+        try:
+            # Upload to github
+            with open(pipeline_filepath) as input_file:
+                content = input_file.read()
+
+                self.repo.create_file(path=pipeline_name + ".py",
+                                      message="Pushed DAG " + pipeline_name,
+                                      content=content,
+                                      branch=self.branch)
+
+            self.log.info('Pipeline successfully added to the git queue')
+
+        except FileNotFoundError as fnfe:
+            self.log.error(f'Unable to locate local DAG file to upload: {pipeline_filepath}: ' + str(fnfe))
+            raise RuntimeError(f'Unable to locate local DAG file to upload: {pipeline_filepath}: ', fnfe)
+        except GithubException as e:
+            self.log.error(f'Error uploading DAG to branch {self.branch}: ' + str(e))
```

Comment:
Same for these two exception handlers.  This prevents this kind of messaging from occurring:
```
During handling of the above exception, another exception occurred:
```
and produces this instead (indicating the extra exception is not really a bug. :smile:
```
The above exception was the direct cause of the following exception:
```

Reply:
done. looks better
![image](https://user-images.githubusercontent.com/13340013/108901128-56362080-75cf-11eb-89dd-da642586c06e.png)